jupyter_core/command.py

```diff
@@ -113,9 +115,24 @@ def _execvp(cmd, argv):
         os.execvp(cmd, argv)
 
 
+def _jupyter_realpath(path, subcommand):
+    # get currrent PATH
+    _path_env = os.environ['PATH']
+    # set new PATH with self
+    os.environ['PATH'] = os.pathsep.join(_path_with_self())
```

Comment:
`shutil.which` takes a `path=` parameter to override the environment variable, so we don't even need to modify the environment variable.